### PR TITLE
Issue #311 auto expanded navigation property

### PR DESCRIPTION
**Pull request by VikingsFan**
Add [AutoExpended] attribute to support auto expanded navigation property

### CODE REVIEW

File: OData/src/System.Web.OData/OData/Builder/AutoExpandedAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using Microsoft.OData.Edm;
+
+namespace System.Web.OData.Builder
+{
+    /// <summary>
+    /// Represents an <see cref="Attribute"/> that can be placed on a navigation property to specify it
+    /// is auto expanded.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Property)]
+    public sealed class AutoExpandedAttribute : Attribute
```
Comment (xuzhg) [inline OData/src/System.Web.OData/OData/Builder/AutoExpandedAttribute.cs]: AutoExpandAttribute or Just

ExpandAttribute

File: OData/src/System.Web.OData/OData/Builder/NavigationPropertyConfiguration.cs [hunk 2]
```diff
@@ -96,6 +96,11 @@ public override PropertyKind Kind
         public EdmOnDeleteAction OnDeleteAction { get; set; }
 
         /// <summary>
+        /// Gets or sets the the navigation property is auto expanded.
+        /// </summary>
+        public bool AutoExpanded { get; set; }
```
Comment (xuzhg) [inline OData/src/System.Web.OData/OData/Builder/NavigationPropertyConfiguration.cs]: public bool Expand {get;set;}

File: OData/src/System.Web.OData/OData/Builder/ODataConventionModelBuilder.cs [hunk 3]
```diff
@@ -622,6 +623,12 @@ private void MapEntityType(EntityTypeConfiguration entity)
                             addedNavigationProperty.Contained();
                         }
 
+                        AutoExpandedAttribute autoExpandedAttribute = property.GetCustomAttribute<AutoExpandedAttribute>();
```
Comment (xuzhg) [inline OData/src/System.Web.OData/OData/Builder/ODataConventionModelBuilder.cs]: Why add these codes here?

You insert the convention into the list, it should be apply automatically.

**Comment by VikingsFan** [comment]
checked in d21a410fb1b36b9455796dd18d5bbdcf10c2f5ad by Liang, Thanks

**Comment by congysu** [comment]
Issue: #311

**Pull request closed without merging**